app/routes/index/mygoals.js

```diff
@@ -2,7 +2,22 @@ import Ember from 'ember';
 import AuthenticatedRouteMixin from 'feedbackfruits-accounts/mixins/routes/authenticated-route-mixin';
 
 export default Ember.Route.extend(AuthenticatedRouteMixin, {
+
   model(){
-    return this.get('store').findAll('goal');
+    /* jshint ignore:start */
+    return this.get('store').findRecord('user','current')
+      .then(I => {
+        let mapFn = (goal) => {
```

Comment:
This can be refactored into something shorter and easier to understand I believe. Furthermore, the contributor_ids are already present and do not need to be fetched I believe.